.github/workflows/test_branch.yml

```diff
@@ -0,0 +1,47 @@
+name: Test Branch
```

Comment:
Can we merge these two files into one? They are almost exact copies of each other.
I think we can add another `on` rule below for PRs and pass `HAZELCAST_ENTERPRISE_KEY` conditionally as an environment variable. WDTY?

Reply:
Makes sense, let me try